### PR TITLE
Add express style router sugar (aka. router.get, router.post, etc)

### PR DESCRIPTION
**Pull request by andyday**
Adding express-style router. get, post, put, etc.  The intent is to simplify usage.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ build/
 demo/out/
 out/
 xrpc.iml
+.DS_Store
```
Comment (jkinkead) [inline .gitignore]: This should go in your global gitignore.

File: src/main/java/com/nordstrom/xrpc/server/Router.java [hunk 2]
```diff
@@ -126,17 +131,50 @@ public void scheduleHealthChecks(
...
-            healthCheckRegistry.runHealthChecks(workerGroup);
-          }
-        },
-        initialDelay,
-        delay,
-        timeUnit);
+        () -> healthCheckRegistry.runHealthChecks(workerGroup), initialDelay, delay, timeUnit);
+  }
+
+  /** * Convenience Methods ** */
```
Comment (jkinkead) [inline src/main/java/com/nordstrom/xrpc/server/Router.java]: Shouldn't be javadoc-formatted.

File: src/main/java/com/nordstrom/xrpc/server/Router.java [hunk 3]
```diff
@@ -126,17 +131,50 @@ public void scheduleHealthChecks(
...
+  }
+
+  public void connect(String route, Handler handler) {
+    addRoute(route, handler, XHttpMethod.CONNECT);
+  }
+
+  public void any(String route, Handler handler) {
+    addRoute(route, handler, XHttpMethod.ANY);
+  }
+  /** ************************* */
```
Comment (jkinkead) [inline src/main/java/com/nordstrom/xrpc/server/Router.java]: Shouldn't be javadoc-formatted.

File: src/main/java/com/nordstrom/xrpc/server/Router.java [hunk 4]
```diff
@@ -126,17 +131,50 @@ public void scheduleHealthChecks(
...
-          }
-        },
-        initialDelay,
-        delay,
-        timeUnit);
+        () -> healthCheckRegistry.runHealthChecks(workerGroup), initialDelay, delay, timeUnit);
+  }
+
+  /** * Convenience Methods ** */
+  public void get(String route, Handler handler) {
```
Comment (jkinkead) [inline src/main/java/com/nordstrom/xrpc/server/Router.java]: It wouldn't hurt to add very simple javadoc to these, especially for ANY:

```java
/** Binds a default handler for all HTTP requests to the given route. */
...
/** Binds a handler for GET requests to the given route. */
```

File: .gitignore (hunk 1 above)
Comment (andyday) [reply .gitignore]: i disagree.   project files should do as much as possible to make dev on-boarding simple.  the dev cost savings over time adds up...

File: src/main/java/com/nordstrom/xrpc/server/Router.java (hunk 2 above)
Comment (andyday) [reply src/main/java/com/nordstrom/xrpc/server/Router.java]: duh... the google java formatter bastardized it...  will fix...

File: src/main/java/com/nordstrom/xrpc/server/Router.java (hunk 4 above)
Comment (andyday) [reply src/main/java/com/nordstrom/xrpc/server/Router.java]: will do

File: src/main/java/com/nordstrom/xrpc/server/Router.java (hunk 3 above)
Comment (andyday) [reply src/main/java/com/nordstrom/xrpc/server/Router.java]: see above...

**Comment by forplayplay** [comment]
I think these helper methods are nice and simple!

I have mixed feelings about definition and registration all happening in main. It's great for small apps - but as they grow, I can see this pattern becoming cumbersome - especially for maintainers on-boarding further down the road. 

I would encourage us to separate handlers by resource in their own controller class. Is there some rational I am missing?

File: src/main/java/com/nordstrom/xrpc/server/Router.java (hunk 2 above)
Comment (andyday) [reply src/main/java/com/nordstrom/xrpc/server/Router.java]: fixed

File: src/main/java/com/nordstrom/xrpc/server/Router.java (hunk 4 above)
Comment (andyday) [reply src/main/java/com/nordstrom/xrpc/server/Router.java]: done

File: src/main/java/com/nordstrom/xrpc/server/Router.java (hunk 3 above)
Comment (andyday) [reply src/main/java/com/nordstrom/xrpc/server/Router.java]: done

**Comment by andyday** [comment]
i've updated based on review.   @forplayplay can you take a look at the revised example.  it shows a way to split routes based on resource group...

File: .gitignore (hunk 1 above)
Comment (jkinkead) [reply .gitignore]: I'm not saying it shouldn't go here; I'm saying that you should have this in your global gitignore so you don't need to add it. :)

File: demo/src/main/java/com/nordstrom/xrpc/demo/PeopleRoutes.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.AllArgsConstructor;
+
+public class PeopleRoutes {
+  final List<Person> people = new ArrayList<>();
+
+  public PeopleRoutes(Router router) {
+    // For the JSON portion of the demo
```
Comment (jkinkead) [inline demo/src/main/java/com/nordstrom/xrpc/demo/PeopleRoutes.java]: We should use Jackson; I'll open an issue to track this.

File: .gitignore (hunk 1 above)
Comment (andyday) [reply .gitignore]: agreed.   added to global.  the one caveat to that is my adding it to global blinds me to the need for it in the project.  i've lead enough teams with junior devs that might mis ignoring such files/dirs and check them in.  having it in the project protects from such things...

File: demo/src/main/java/com/nordstrom/xrpc/demo/PeopleRoutes.java (hunk 5 above)
Comment (andyday) [reply demo/src/main/java/com/nordstrom/xrpc/demo/PeopleRoutes.java]: yep.  will be doing this in my next pr.

**Pull request merged**